taf/repository_tool.py

```diff
@@ -0,0 +1,233 @@
+import getpass
+import json
+import os
+import shutil
+from contextlib import contextmanager
+from pathlib import Path
+
+import securesystemslib
+
+role_keys_cache = {}
+
+
+class Repository:
+
+  def __init__(self, repository, repository_path):
+    self._repository = repository
+    self.repository_path = repository_path
+
+  _required_files = ['repositories.json']
+
+  @property
+  def targets_path(self):
+    return os.path.join(self.repository_path, 'targets')
+
+  @property
+  def metadata_path(self):
+    return os.path.join(self.repository_path, 'metadata')
+
+  @property
+  def metadata_staged_path(self):
+    return os.path.join(self.repository_path, 'metadata.staged')
+
+  def add_existing_target(self, file_path, targets_role='targets'):
+    """
+    Registers new target files with TUF. The files are expected to be
+    inside the targets directory.
+    """
+    targets = self._role_obj(targets_role)
+    targets.add_target(os.path.join(self.targets_path, file_path))
+
+  def add_targets(self, data, targets_role='targets', files_to_keep=[]):
```

Comment:
Please change `files_to_keep` to `None` default value instead of `[]`, because the _same_ default list will be used for every `add_targets` call, if `files_to_keep` argument is not passed.

```python
required_files = ['test']

def test(files_to_keep=[]):
  if files_to_keep is None:
    files_to_keep = []
  files_to_keep.extend(required_files)
  print(files_to_keep)

test()
test()
```

Output is:

```
['test']
['test', 'test']
```

Reply:
Good catch. I forgot about that. Fixed